velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,321 @@
+                 {"n_nationkey", "n_name"},
+                 1 /*scaleFactor*/)
+             .planNode();
+
+  auto nations = AssertQueryBuilder(plan).split(makeTpchSplit()).copyResults(pool());
+
+  std::cout << std::endl
+            << "> first 10 rows from TPC-H nation table: "
+            << nations->toString() << std::endl;
+  std::cout << nations->toString(0, 10) << std::endl;
+
+.. code-block::
+
+    0: {0, ALGERIA}
+    1: {1, ARGENTINA}
+    2: {2, BRAZIL}
+    3: {3, CANADA}
+    4: {4, EGYPT}
+    5: {5, ETHIOPIA}
+    6: {6, FRANCE}
+    7: {7, GERMANY}
+    8: {8, INDIA}
+    9: {9, INDONESIA}
+
+Joins
+~~~~~
+
+We can now join TPC-H nation and region tables to count number of nations in
+each region and sort results by region name. We need to use one TableScan node
+for nations table and another for region table. We also need to provide splits
+for each TableScan node. We will use two PlanBuilders: one for the probe
+side of the join and another one for the build side. We will also use
+PlanNodeIdGenerator to ensure that all plan nodes in the final plan have unique
+IDs.
+
+.. code-block:: c++
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId nationScanId;
+  core::PlanNodeId regionScanId;
+  plan = PlanBuilder(planNodeIdGenerator)
+             .tableScan(
+                 tpch::Table::TBL_NATION, {"n_regionkey"}, 1 /*scaleFactor*/)
+             .capturePlanNodeId(nationScanId)
+             .hashJoin(
+                 {"n_regionkey"},
+                 {"r_regionkey"},
+                 PlanBuilder(planNodeIdGenerator)
+                     .tableScan(
+                         tpch::Table::TBL_REGION,
+                         {"r_regionkey", "r_name"},
+                         1 /*scaleFactor*/)
+                     .capturePlanNodeId(regionScanId)
+                     .planNode(),
+                 "", // extra filter
+                 {"r_name"})
+             .singleAggregation({"r_name"}, {"count(1) as nation_cnt"})
+             .orderBy({"r_name"}, false)
+             .planNode();
+
+  auto nationCnt = AssertQueryBuilder(plan)
+                       .split(nationScanId, makeTpchSplit())
+                       .split(regionScanId, makeTpchSplit())
+                       .copyResults(pool());
+
+  std::cout << std::endl
+            << "> number of nations per region in TPC-H: "
+            << nationCnt->toString() << std::endl;
+  std::cout << nationCnt->toString(0, 10) << std::endl;
+
+.. code-block::
+
+    0: {AFRICA, 5}
+    1: {AMERICA, 5}
+    2: {ASIA, 5}
+    3: {EUROPE, 5}
+    4: {MIDDLE EAST, 5}
+
+Check out :doc:`Joins <develop/joins>` to learn more about joins and
+join-specific optimizations in Velox. Also, you can find a list of
+available plan nodes in :doc:`Plan Nodes and Operators <develop/operators>`
+or in PlanNodeId.h and PlanBuilder.h files in the code.
+
+Curious about reading data from a Hive connector? Check out
```

Comment:
Move this sentence to the Connectors section ?

Reply:
I'd like to keep this at the end to avoid distracting the reader.